stats: respect turbo cheat

rr- opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

Resolves #2167.

Now that 5785b0e65 has eliminated the clock drift problems, we can return to simple frame-based timer increments.
I've compared the current and proposed approaches for normal game speed. There are some initial differences of up to 4 frames due to loading times, but they remain stable and don't increase over time.
I kept the original code behind `#ifdef` but I think we maybe should remove it after all. I'd like to defer the final decision to @lahm86.

rr- committed:
stats: respect turbo cheat

Resolves #2167.

Now that 5785b0e65 has eliminated the clock drift problems, we can
return to simple frame-based timer increments.

lahm86 reviewed (approved):
LGTM.
I think it's fine to remove the previous code, we always have the repo history if we need to consult back for any reason.

aredfan commented:
Issue in TR2X:

- [x] Turbo speed 2 isn't faster than 1 when used for the game's stopwatch in the story levels, for example Floating Islands. The assault course timer isn't affected by this issue.

rr- committed:
tr2/inv-ring: fix frames counted wrongly

Inventory needs to be animated twice per frame, but the way we went
about it, caused some issues - for example, the ingame timer would be
incremented wrong number of times, or the textures would animate at
wrong speed (not that we can currently see them).

rr- commented:
@aredfan fixed.
